### PR TITLE
fix bug when combining groupby with resample and interpolate with dat…

### LINKED ISSUE #35325: BUG: Combination of groupby.resample.interpolate() fails
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [x] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample, a copy-pastable example

```python
d = {'price': [10, 11, 9, 13, 14, 18, 17, 19],
     'volume': [50, 60, 40, 100, 50, 100, 40, 50]}

df = pd.DataFrame(d)

df['week_starting'] = pd.date_range('01/01/2018', periods=8, freq='W')

df \
    .set_index("week_starting") \
    .groupby("volume") \
    .resample("1D") \
    .interpolate(method="linear")

```

Error:
```
TypeError                                 Traceback (most recent call last)
~/.cache/pypoetry/virtualenvs/lagps-6BuXYM4Y-py3.8/lib/python3.8/site-packages/pandas/core/groupby/groupby.py in apply(self, func, *args, **kwargs)
    735             try:
--> 736                 result = self._python_apply_general(f)
    737             except TypeError:

~/.cache/pypoetry/virtualenvs/lagps-6BuXYM4Y-py3.8/lib/python3.8/site-packages/pandas/core/groupby/groupby.py in _python_apply_general(self, f)
    751     def _python_apply_general(self, f):
--> 752         keys, values, mutated = self.grouper.apply(f, self._selected_obj, self.axis)
    753 

~/.cache/pypoetry/virtualenvs/lagps-6BuXYM4Y-py3.8/lib/python3.8/site-packages/pandas/core/groupby/ops.py in apply(self, f, data, axis)
    205             group_axes = group.axes
--> 206             res = f(group)
    207             if not _is_indexed_like(res, group_axes):

~/.cache/pypoetry/virtualenvs/lagps-6BuXYM4Y-py3.8/lib/python3.8/site-packages/pandas/core/resample.py in func(x)
    988 
--> 989             return x.apply(f, *args, **kwargs)
    990 

~/.cache/pypoetry/virtualenvs/lagps-6BuXYM4Y-py3.8/lib/python3.8/site-packages/pandas/core/resample.py in aggregate(self, func, *args, **kwargs)
    284             grouper = None
--> 285             result = self._groupby_and_aggregate(how, grouper, *args, **kwargs)
    286 

~/.cache/pypoetry/virtualenvs/lagps-6BuXYM4Y-py3.8/lib/python3.8/site-packages/pandas/core/resample.py in _groupby_and_aggregate(self, how, grouper, *args, **kwargs)
    360             else:
--> 361                 result = grouped.aggregate(how, *args, **kwargs)
    362         except DataError:

~/.cache/pypoetry/virtualenvs/lagps-6BuXYM4Y-py3.8/lib/python3.8/site-packages/pandas/core/groupby/generic.py in aggregate(self, func, *args, **kwargs)
    923             # nicer error message
--> 924             raise TypeError("Must provide 'func' or tuples of '(column, aggfunc).")
    925 

TypeError: Must provide 'func' or tuples of '(column, aggfunc).

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
<ipython-input-15-5d08bff36e4d> in <module>
----> 1 df = df \
      2     .set_index("week_starting") \
      3         .groupby("volume") \
      4             .resample("1D") \
      5                 .interpolate(method="linear")

~/.cache/pypoetry/virtualenvs/lagps-6BuXYM4Y-py3.8/lib/python3.8/site-packages/pandas/core/resample.py in interpolate(self, method, axis, limit, inplace, limit_direction, limit_area, downcast, **kwargs)
    797         Interpolate values according to different methods.
    798         """
--> 799         result = self._upsample(None)
    800         return result.interpolate(
    801             method=method,

~/.cache/pypoetry/virtualenvs/lagps-6BuXYM4Y-py3.8/lib/python3.8/site-packages/pandas/core/resample.py in _apply(self, f, grouper, *args, **kwargs)
    989             return x.apply(f, *args, **kwargs)
    990 
--> 991         result = self._groupby.apply(func)
    992         return self._wrap_result(result)
    993 

~/.cache/pypoetry/virtualenvs/lagps-6BuXYM4Y-py3.8/lib/python3.8/site-packages/pandas/core/groupby/groupby.py in apply(self, func, *args, **kwargs)
    745 
    746                 with _group_selection_context(self):
--> 747                     return self._python_apply_general(f)
    748 
    749         return result

~/.cache/pypoetry/virtualenvs/lagps-6BuXYM4Y-py3.8/lib/python3.8/site-packages/pandas/core/groupby/groupby.py in _python_apply_general(self, f)
    750 
    751     def _python_apply_general(self, f):
--> 752         keys, values, mutated = self.grouper.apply(f, self._selected_obj, self.axis)
    753 
    754         return self._wrap_applied_output(

~/.cache/pypoetry/virtualenvs/lagps-6BuXYM4Y-py3.8/lib/python3.8/site-packages/pandas/core/groupby/ops.py in apply(self, f, data, axis)
    204             # group might be modified
    205             group_axes = group.axes
--> 206             res = f(group)
    207             if not _is_indexed_like(res, group_axes):
    208                 mutated = True

~/.cache/pypoetry/virtualenvs/lagps-6BuXYM4Y-py3.8/lib/python3.8/site-packages/pandas/core/resample.py in func(x)
    987                 return getattr(x, f)(**kwargs)
    988 
--> 989             return x.apply(f, *args, **kwargs)
    990 
    991         result = self._groupby.apply(func)

~/.cache/pypoetry/virtualenvs/lagps-6BuXYM4Y-py3.8/lib/python3.8/site-packages/pandas/core/resample.py in aggregate(self, func, *args, **kwargs)
    283             how = func
    284             grouper = None
--> 285             result = self._groupby_and_aggregate(how, grouper, *args, **kwargs)
    286 
    287         result = self._apply_loffset(result)

~/.cache/pypoetry/virtualenvs/lagps-6BuXYM4Y-py3.8/lib/python3.8/site-packages/pandas/core/resample.py in _groupby_and_aggregate(self, how, grouper, *args, **kwargs)
    359                 result = grouped._aggregate_item_by_item(how, *args, **kwargs)
    360             else:
--> 361                 result = grouped.aggregate(how, *args, **kwargs)
    362         except DataError:
    363             # we have a non-reducing function; try to evaluate

~/.cache/pypoetry/virtualenvs/lagps-6BuXYM4Y-py3.8/lib/python3.8/site-packages/pandas/core/groupby/generic.py in aggregate(self, func, *args, **kwargs)
    922         elif func is None:
    923             # nicer error message
--> 924             raise TypeError("Must provide 'func' or tuples of '(column, aggfunc).")
    925 
    926         func = _maybe_mangle_lambdas(func)

TypeError: Must provide 'func' or tuples of '(column, aggfunc).
```

On master the error is raised in line 86 of the same file.

#### Problem description

The combination of `groupby`, `resample`, and `interpolate` leads to an `TypeError: Must provide 'func' or tuples of '(column, aggfunc).` [0].

Other functions like `ffill`, or `bfill` work without issues.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.8.2.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.4.0-18362-Microsoft
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : C.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.0.5
numpy            : 1.19.0
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 20.0.2
setuptools       : 44.0.0
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : 2.8.5 (dt dec pq3 ext lo64)
jinja2           : 2.11.2
IPython          : 7.16.1
pandas_datareader: None
bs4              : None
bottleneck       : 1.3.2
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : 3.2.2
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
pytest           : None
pyxlsb           : None
s3fs             : None
scipy            : 1.5.0
sqlalchemy       : 1.3.18
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None
numba            : None

</details>

[0], btw, there is a quote missing in the error message.

### PR DESCRIPTION
**Pull request by CloseChoice**
…etime-index (GH 35325)

- [x] closes #35325
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Review by arw2019 — APPROVED** [review]
lgtm

**Comment by simonjayhawkins** [comment]
Thanks @CloseChoice for the PR. The test failures should be fixed on master. can you merge upstream/master.

**Comment by CloseChoice** [comment]
> Thanks @CloseChoice for the PR. The test failures should be fixed on master. can you merge upstream/master.

Done, though I am not quite sure if my whatsnew entry is correct. There seemed to be some problems with building the documentation. I also had a similar issue on another PR.

File: doc/source/whatsnew/v1.1.0.rst [hunk 1]
```diff
@@ -1132,6 +1132,7 @@ Groupby/resample/rolling
 - Bug in :meth:`core.groupby.DataFrameGroupBy.transform` when ``func='nunique'`` and columns are of type ``datetime64``, the result would also be of type ``datetime64`` instead of ``int64`` (:issue:`35109`)
 - Bug in :meth:`DataFrame.groupby` raising an ``AttributeError`` when selecting a column and aggregating with ``as_index=False`` (:issue:`35246`).
 - Bug in :meth:'DataFrameGroupBy.first' and :meth:'DataFrameGroupBy.last' that would raise an unnecessary ``ValueError`` when grouping on multiple ``Categoricals`` (:issue:`34951`)
+- Bug when combining methods :meth:`DataFrame.groupby` with :meth:`DataFrame.resample` and :meth:`DataFrame.interpolate` raising an ``TypeError`` (:issue:`35325`)
```
Comment (arw2019) [inline doc/source/whatsnew/v1.1.0.rst:1135]: I might do 
```
- Bug in `meth`:`Resampler.interpolate` when combining :meth:`DataFrame.groupby`, :meth:`DataFrame.resample` and :meth:`DataFrame.interpolate` was raising a ``TypeError`` (:issue:`35325`)
```

**Review by arw2019 — COMMENTED** [review]
Release note might need to be moved to 1.2 (see #34730)

File: doc/source/whatsnew/v1.1.0.rst (hunk 1 above)
Comment (CloseChoice) [reply doc/source/whatsnew/v1.1.0.rst:1135]: > Release note might need to be moved to 1.2 (see #34730)

Thanks for the link. I was looking for the whatsnew 1.2 but have seen that the PR, which will add this, is still open.

Comment (simonjayhawkins) [reply doc/source/whatsnew/v1.1.0.rst:1135]: has now been merged

File: pandas/tests/resample/test_resampler_grouper.py [hunk 2]
```diff
@@ -346,3 +346,65 @@ def test_median_duplicate_columns():
     result = df.resample("5s").median()
     expected.columns = result.columns
     tm.assert_frame_equal(result, expected)
+
```
Comment (jreback) [inline pandas/tests/resample/test_resampler_grouper.py:349]: can you move this to pandas/tests/resample/test_time_grouper.py

**Review by jreback — CHANGES_REQUESTED** [review]
looks ok, can you merge master and ping on green (after change)

File: doc/source/whatsnew/v1.2.0.rst [hunk 3]
```diff
@@ -152,10 +152,14 @@ Plotting
 Groupby/resample/rolling
 ^^^^^^^^^^^^^^^^^^^^^^^^
 
+<<<<<<< HEAD
```
Comment (jreback) [inline doc/source/whatsnew/v1.2.0.rst:155]: can you remove the rebase markers

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v1.2.0.rst (hunk 3 above)
Comment (CloseChoice) [reply doc/source/whatsnew/v1.2.0.rst]: done

**Comment by CloseChoice** [comment]
> looks ok, can you merge master and ping on green (after change)

@jreback Friendly ping ;)

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @CloseChoice very nice

**Pull request merged**